src/engine/guidance/post_processing.cpp

```diff
@@ -1183,7 +1193,9 @@ std::vector<RouteStep> collapseUseLane(std::vector<RouteStep> steps)
         if (step.maneuver.instruction.type == TurnType::UseLane &&
             canCollapeUseLane(step.maneuver.lanes, step.maneuver.lane_description))
         {
-            elongate(steps[step_index - 1], steps[step_index]);
+            const auto previous = getPreviousIndex(step_index);
+            steps[previous] = elongate(steps[previous], steps[step_index]);
```

Comment:
Argh why is there no warning for us ignoring the return value before? There is

    __attribute__((warn_unused_result))

but man this should be on by default..

This needs to go in as soon as possible.